tests/test_kadet.py

```diff
@@ -0,0 +1,162 @@
+#!/usr/bin/env python3
+#
+# Copyright 2018 The Kapitan Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"kadet tests"
+
+import tempfile
+import unittest
+from kapitan.inputs.kadet import BaseObj, Dict
+
+
+class KadetTestObj(BaseObj):
+    def new(self):
+        self.need('name', 'Need a name string')
+        self.need('size', 'Need a size int')
+
+    def body(self):
+        self.root.name = self.kwargs.name
+        self.root.size = self.kwargs.size
+        self.root.firstkey = 1
+        self.root.nested.firstkey = 2
+        self.root['tradicional_key'] = 3
+        self.root.withdict = {'A': 'dict'}
+        self.root.withbaseobj_init_as = BaseObj.from_dict({'init': 'as'})
+        bobj = BaseObj()
+        bobj.root.inside = "BaseObj"
+        self.root.withbaseobj = bobj
```

Comment:
Some of the vars here under self.root are oneword, some are oneWord and some are oneword_and_dashes. 
Would it be more readable if we followed one convention? Something like `self.root.i_am_inside` used below seems nice. :)